cpp/src/transform/row_bit_count.cu

```diff
@@ -0,0 +1,508 @@
+                      h_info,
+                      stream,
+                      cur_depth + 1,
+                      cur_branch_depth,
+                      out.size() - 1);
+  }
+
+  // everything else
+  template <typename T,
+            std::enable_if_t<!cudf::is_fixed_width<T>() && !std::is_same<T, string_view>::value &&
+                             !std::is_same<T, list_view>::value &&
+                             !std::is_same<T, struct_view>::value>* = nullptr>
+  void operator()(column_view const& col,
+                  std::vector<cudf::column_view>& out,
+                  std::vector<column_info>& info,
+                  hierarchy_info& h_info,
+                  rmm::cuda_stream_view stream,
+                  size_type cur_depth,
+                  size_type cur_branch_depth,
+                  int parent_index)
+  {
+    CUDF_FAIL("Unsupported column type in row_bit_count");
+  }
+};
+
+template <typename ColIter>
+void flatten_hierarchy(ColIter begin,
+                       ColIter end,
+                       std::vector<cudf::column_view>& out,
+                       std::vector<column_info>& info,
+                       hierarchy_info& h_info,
+                       rmm::cuda_stream_view stream,
+                       size_type cur_depth,
+                       size_type cur_branch_depth,
+                       int parent_index)
+{
+  std::for_each(begin, end, [&](column_view const& col) {
+    cudf::type_dispatcher(col.type(),
+                          flatten_functor{stream},
+                          col,
+                          out,
+                          info,
+                          h_info,
+                          stream,
+                          cur_depth,
+                          cur_branch_depth,
+                          parent_index);
+  });
+}
+
+/**
+ * @brief Struct representing a span of rows.
+ *
+ */
+struct row_span {
+  size_type row_start, row_end;
+};
+
+/**
+ * @brief Functor for computing the size, in bits, of a `row_span` of rows for a given
+ * `column_device_view`
+ *
+ */
+struct row_size_functor {
+  template <typename T>
+  __device__ size_type operator()(column_device_view const& col, row_span const& span)
+  {
+    auto const num_rows{span.row_end - span.row_start};
+    auto const element_size  = sizeof(device_storage_type_t<T>) * 8;
+    auto const validity_size = col.nullable() ? 1 : 0;
+    return (element_size + validity_size) * num_rows;
+  }
+};
+
+template <>
+__device__ size_type row_size_functor::operator()<string_view>(column_device_view const& col,
+                                                               row_span const& span)
+{
+  column_device_view const& offsets = col.child(strings_column_view::offsets_column_index);
+  auto const num_rows{span.row_end - span.row_start};
+  auto const row_start{span.row_start + col.offset()};
+  auto const row_end{span.row_end + col.offset()};
+
+  auto const offsets_size  = sizeof(offset_type) * 8;
```

Comment:
```suggestion
  auto const offsets_size  = sizeof(offset_type) * CHAR_BIT;
```
There is `CHAR_BIT` defined in `<limits.h>` that we can use to avoid the magic digits.
I know that `* 8` is quite prevalent already in libcudf.